### PR TITLE
layout: Do not propagate floats in or out of absolutely positioned flows

### LINKED ISSUE #10625: Absolutely-positioned inlines don't respect line height if positioned next to a float
Fun one. Failing test case:

    <!DOCTYPE html>
    <div style="float: left; padding: 0 0 0 1px;"></div>
    <span style="position: absolute; right: 0; top: 0; line-height: 100px;">X</span>

This should have an X positioned 100px down the page on the right side, but in Servo the X is at the top right corner of the page.

Reduced from http://reddit.com/r/rust (the Edit link at the top).

### PR DESCRIPTION
**Pull request by notriddle**
Fixes #10625

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10637)

### CODE REVIEW

File: components/layout/sequential.rs [hunk 1]
```diff
@@ -144,10 +145,15 @@ pub fn guess_float_placement(flow: &mut Flow) {
 
     let mut floats_in = SpeculatedFloatPlacement::compute_floats_in_for_first_child(flow);
     for kid in flow::mut_base(flow).child_iter_mut() {
-        floats_in.compute_floats_in(kid);
-        flow::mut_base(kid).speculated_float_placement_in = floats_in;
-        guess_float_placement(kid);
-        floats_in = flow::base(kid).speculated_float_placement_out;
+        if flow::base(kid).flags.contains(IS_ABSOLUTELY_POSITIONED) {
+            // Do not propogate floats in or out, but do propogate between kids.
```
Comment (pcwalton) [inline components/layout/sequential.rs]: typo: "propogate" -> "propagate"

**Comment by pcwalton** [comment]
Looks good modulo that typo, but needs a reftest.

**Comment by pcwalton** [comment]
Fix the same typo in the commit message and then r=me. :)

**Comment by notriddle** [comment]
Typo in commit message has now been fixed.

**Pull request merged**